media_service/iiif.py

```diff
@@ -208,6 +208,11 @@ def __init__(self, manifest, sequence_id):
         self.canvases = []
 
     def add_canvas(self, canvas_id):
+        # Mirador relies on every canvas having a unique URI, so in case we have duplicate image
+        # objects in the manifest (i.e. duplicate canvases), we'll add an index to each canvas_id
+        # to ensure uniqueness within the manifest.
+        index = len(self.canvases)
+        canvas_id = "%s.%s" % (canvas_id, index)
```

Comment:
If images are added, then some removed (I don't know if this is possible in this app), then a 'duplicate' is added, is it possible that we'll get name clashes still?

e.g.
image a added
image b added
image b added again (would be `b3`?)
image a removed
image b added again (would try to add `b3`?)